spec/parsers/arw_parser_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'spec_helper'
+
+describe FormatParser::ARWParser do
+  describe 'matches filenames with valid extensions' do
+    filenames = ['raw_file', 'another raw file', 'and.another', 'one-more']
+    extensions = ['.arw', '.Arw', '.aRw', '.arW', '.ARw', '.ArW', '.aRW', '.ARW']
+    filenames.each do |filename|
```

Comment:
Maybe using rspec's `include_examples` and `shared_examples` could make this a bit cleaner :)

Reply:
I'm not familiar with shared_examples (I had a quick look and it looks like it can factor out repeated test code especially for a type or module https://relishapp.com/rspec/rspec-core/v/2-14/docs/example-groups/shared-examples) but what would you like me to clean up here?

The cross join between the 2 collections to get the test file names? As the assertion is different between the 2 tests factoring out the entire test code seems more work than it would save but maybe I'm miss-understanding what you are suggesting.

Reply:
We don't use it a lot throughout the code base, but `aac_parser_spec.rb` and a few other places benefit from it.

Even if you don't use these examples it across multiple describe blocks, it still gives you a descriptive way to setup examples for your multiple `it` blocks. Using it, you don't need to explicitly call `.each` and it makes it easier to come up with a `describe` block that won't look like an `it`.

The way you have it is definitely fine. I bring it up because I found it neat when I saw the guys using `shared_examples`.

Reply:
Thanks for explaining it and pointing me at some samples.

I've added some extra tests and cleaned up some of the existing likely_match tests to use shared_examples.

I also added some error cases as shared examples to confirm that other file formats do not get parsed.